staging/src/k8s.io/apimachinery/pkg/api/meta/interfaces.go

```diff
@@ -109,7 +111,7 @@ type RESTMapping struct {
 // to API groups. In other words, kinds and resources should not be assumed to be
 // unique across groups.
 //
-// TODO: split into sub-interfaces
+// Deprecated: use RESTMapperWithContext instead.
```

Comment:
This applies to other files as well. I remember we agreed NOT to deprecate these, but rather put a message like:
```
// Contextual logging: RESTMapperWithContext should be used instead of RESTMapper in code which supports contextual logging.
```

Reply:
That was for functions where the original version is functionally sufficient for its intended purpose.

Here we have a case of an interface that looks like it works locally (no context) but under the hood often must call blocking functions which really should be called with a context if the user of this interface has one. I think this is a case where we should nudge the ecosystem towards updating their code by declaring the old functions as deprecated.